lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -462,12 +465,28 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    def tags_from_namespace_labels
+      namespace_info = nil
+      with_retries(2) do
+        namespace_info, _, st = kubectl.run("get", "namespace", @namespace, "-o", "json", use_namespace: false,
+          log_failure: true)
+        st.success?
+      end
+      tags = []
```

Comment:
I think that this line and following could be more concisely expressed as:
```ruby
  return [] if namespace_info.blank?
  namespace_labels = JSON.parse(namespace_info, symbolize_names: true).fetch(:metadata, {}).fetch(:labels, {})
  namespace_labels.map { |key, value| "#{key}:#{value}" }
```
Just a suggestion. I don't feel strongly about it.